src/workerd/api/global-scope.c++

```diff
@@ -459,8 +459,28 @@ jsg::Promise<void> ServiceWorkerGlobalScope::test(
   return testHandler(lock, jsg::alloc<TestController>(), eh.env.addRef(lock), eh.getCtx());
 }
 
+kj::Promise<void> ServiceWorkerGlobalScope::setHibernatableEventTimeout(kj::Promise<void> original, kj::Maybe<uint32_t> eventTimeout) {
+  // If we have a maximum event duration timeout set, we should prevent the actor from running
+  // for more than the user selected duration.
+  KJ_IF_SOME(timeout, eventTimeout) {
```

Comment:
nit (non-blocking at all, just decreases indentation): similar early return 
```
if (eventTimeout == kj::none) {
  // return original
}

auto timeout = KJ_REQUIRE_NONNULL(eventTimeout);
// do the join
```

Note that we don't need to check if the timeout is greater than 0 because at the JS API level setting a timeout of 0 unsets the timeout (i.e. it's handled in the `kj::none` case).

Reply:
This is because when in process-sandbox, capnp seems to default the value to 0 instead of `kj::none` if unset.

Reply:
Hmmm... in that case maybe we should convert 0 to kj::none when we call `HibernatableSocketParams::setTimeout()`?

Reply:
Doesn't work. Will only accept `uint32_t`

Reply:
You could also simplify this like:

```cpp
auto timeout = eventTimeout.orDefault0);
if (timeout > 0) {
  // ...
}
```